tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/upgrade/animated/InitConfigureProjectPage.java

```diff
@@ -560,8 +560,8 @@ private void _checkProjectType(IProject project) {
 
 		List<IFile> searchFiles = new SearchFilesVisitor().searchFiles(project, "service.xml");
 
-		if (searchFiles.isEmpty()) {
-			dataModel.setHasServiceBuilder(false);
+		if (!searchFiles.isEmpty()) {
+			dataModel.setHasServiceBuilder(true);
```

Comment:
This should be `dataModel.setHasServiceBuilder(searchFiles.size > 0)`

Reply:
Hey Greg, 
I tried your advice. it will still not show the build service page in this way.
I use latam-upgade-workshop as the sample project, it will setHasServiceBuilder to false if one of them is not include service.xml file among the three projects(latam-upgrade-workshop,tasks-portlet,tasks-portlet-service).